### ISSUE TITLE
Parser issue with comment

### ISSUE THREAD

**Issue by darren-k** [kirbysayshi/vash#70]
I found that commenting out this line caused an issue with the parser. Obviously I can work round this but it might be helpful for you to know. The line is as follows but without any spaces (I had to do that to prevent the line being commented out from the issue)

```<!--li><a href="#features">Features</a>< /li-->```

Thanks for all your great work!

**Comment by kirbysayshi** [comment]
@darren-k Thanks for the report! I updated your comment to use code formatting, is that the actual formatting that was causing the error?

**Comment by darren-k** [comment]
Apologies, I should have put error information here as well. No it wasn't the formatting, the code I used looks exactly how you have amended it. If I get a chance later I'll replicate it and post the actual error message for you.

**Comment by kirbysayshi** [comment]
Oh no problem! My previous comment was ambiguous. By "code formatting" I meant markdown code formatting. By "formatting" I meant "does the code as it appears in the first post cause the error?"

**Comment by kirbysayshi** [comment]
@darren-k I published a fix ass v0.8.7, please let me know (and reopen) if this continues to be a problem! HTML comments weren't really handled by the parser before, but they are now.

**Comment by steebchen** [comment]
When I use simple html comments, html elements are rendered in v0.12.2:

```html
@if (true) {
    <!-- <div> this shouldn't be visible </div> -->
    <div>this should be visible</div>
}
```

output:

```
this shouldn't be visible
this should be visible
```

When I remove the div tag in the html comment, the content isn't being rendered, only if there are html elements. Sometimes, I comment some html tags to hide them, but it's a bit frustrating to always use vash comments instead of my simple atom shortcut to toggle html comments. Does this belong to this thread or is this a separate issue?
